src/tests/idle_node_in_the_background.rs

```diff
@@ -0,0 +1,272 @@
+
+    let mut cfg = DevTestCfg::default();
+    cfg.tracing = TracingOpt::On;
+    cfg.crawl = PeriodicCrawlOpt::On(Duration::from_secs(2));
+    node_run_forever(cfg).await;
+
+    panic!("the node shouldn't have died");
+}
+
+#[tokio::test]
+#[allow(non_snake_case)]
+#[ignore = "convenience test to tinker with a running node for dev purposes"]
+async fn dev002_t1_MONITOR_NODE_FOREVER_WITH_SYNTH_NODE_sn_is_conn_initiator() {
+    // This test is used for testing/development purposes.
+
+    let mut cfg = DevTestCfg::default();
+    cfg.tracing = TracingOpt::On;
+    cfg.crawl = PeriodicCrawlOpt::On(Duration::from_secs(5));
+    cfg.synth_node = SynthNodeOpt::On_TryToConnect(TestConfig::default());
+    node_run_forever(cfg).await;
+
+    panic!("the node shouldn't have died");
+}
+
+#[tokio::test]
+#[allow(non_snake_case)]
+#[ignore = "convenience test to tinker with a running node for dev purposes"]
+async fn dev002_t2_MONITOR_NODE_FOREVER_WITH_SYNTH_NODE_sn_is_conn_responder() {
+    // This test is used for testing/development purposes.
+
+    let mut cfg = DevTestCfg::default();
+    cfg.log_to_stdout = NodeLogToStdout::On;
+    cfg.tracing = TracingOpt::On;
+    cfg.crawl = PeriodicCrawlOpt::On(Duration::from_secs(5));
+    cfg.synth_node = SynthNodeOpt::On_OnlyListening(TestConfig::default());
+    node_run_forever(cfg).await;
+
+    panic!("the node shouldn't have died");
+}
+
+/// Runs the node forever!
+/// The test asserts the node process won't be killed.
+///
+/// Function complexity is increased due to many customization options,
+/// which is not nice but it is what it is.
+///
+/// In short, here are the customization options which are provided via the cfg arg:
+///
+///  - enable/disable node's logs to stdout [cfg.log_to_stdout]
+///
+///  - enable/disable tracing [cfg.tracing]
+///
+///  - enable/disable printing crawler response periodically [cfg.crawl]
+///    - suboption: the duration of the period
+///
+///  - enable/disable attaching a single synthetic node to the node [cfg.synth_node]
+///    - suboption: choose the initiator for the connection
+///    - SyntheticNode's TestConfig configuration is customizable
+///
+async fn node_run_forever(cfg: DevTestCfg) {
+    let target = TempDir::new().expect(ERR_TEMPDIR_NEW);
+    let log_to_stdout = cfg.log_to_stdout.is_on();
+
+    // Enable tracing possibly.
+    if cfg.tracing == TracingOpt::On {
+        synth_node::enable_tracing();
+    }
+
+    // SyntheticNode is spawned only if option is chosen in cfg options.
+    let mut initial_peers = vec![];
+    let mut synth_node: Option<SyntheticNode> = match cfg.synth_node {
+        SynthNodeOpt::On_TryToConnect(cfg) => Some(SyntheticNode::new(&cfg).await),
+        SynthNodeOpt::On_OnlyListening(cfg) => {
+            let sn = SyntheticNode::new(&cfg).await;
+            sn.start_listening()
+                .await
+                .expect("a synth node couldn't start listening");
```

Comment:
ERR_SYNTH_START_LISTENING was error code intended to be used with `start_listening()`

Reply:
uh! good catch!

Reply:
Done.